packages/backend/src/core/discovery/utils/processSources.ts

```diff
@@ -0,0 +1,64 @@
+import { EthereumAddress } from '@l2beat/shared'
+import path from 'path'
+import z from 'zod'
+
+export function processSources(
+  code: string,
+  address: EthereumAddress,
+  contractName: string,
+): Record<string, string> {
+  let result: Record<string, string> = {}
+
+  const meta = `Address: ${address.toString()}}\nContract: ${contractName}`
+  result['meta.txt'] = meta
+
+  if (!code.startsWith('{')) {
+    result[`${contractName}.sol`] = code
+  } else {
+    try {
+      result = { ...result, ...parseSource(code) }
+    } catch (e) {
+      console.error(e)
+      console.log(code)
+    }
+  }
+
+  return result
+}
+
+type Sources = z.infer<typeof Sources>
+const Sources = z.record(z.object({ content: z.string() }))
+const EtherscanSource = z.object({ language: z.string(), sources: Sources })
+
+function parseSource(source: string) {
+  if (source.startsWith('{{')) {
```

Comment:
comment will be useful, I assume that sometimes Etherscan returns data in different format?